src/Toolkit/Iterator.php

```diff
@@ -15,6 +15,8 @@
  * @link      https://getkirby.com
  * @copyright Bastian Allgeier
  * @license   https://opensource.org/licenses/MIT
+ *
+ * @psalm-suppress MissingTemplateParam
```

Comment:
What does this do? Or rather, why is this needed?

Reply:
Psalm started throwing this type of error for the `include $autoloader` calls - disregarding that we just check before for `is_file($autoloader)`. I took it as false positive from their new 5.x version. Thought it's the easiest to suppress it here for now.

Reply:
I think you are referring to the other Psalm-related change. I agree with you on that one.

But this one I don't understand.

Reply:
My mind only saw "missing" in the GitHub preview and jumped to the wrong one.

I don't either honestly. It might be new with psalm 5: https://psalm.dev/running_psalm/issues/MissingTemplateParam/ seems like psalm would like us to specify how we exactly implement IteratorAggregate. I wouldn't quite know how that specification should look like. I don't think it gives us much benefits. It has worked without so far. That's why I suppressed it.

Reply:
I looked into it more and fixed it with Psalm annotations.

Reply:
I think we would have to add these to every class that extends Iterator. Not sure that is worth it.

Reply:
You are right, I didn't realize this was the case. I've reverted it and added a comment for future reference.